support/metrics/metricsTracker.go

```diff
@@ -190,34 +188,65 @@ func MakeMetricsTracker(
 	}, nil
 }
 
+// MakeMetricsTrackerGui is a factory method to create a `metrics.Tracker` from the CLI.
+func MakeMetricsTrackerGui(
+	userID string,
+	deviceID string,
+	apiKey string,
+	client *http.Client,
+	botStartTime time.Time,
+	version string,
+	goos string,
+	goarch string,
+	goarm string,
+	guiVersion string,
+) (*MetricsTracker, error) {
+	props := commonProps{
+		CliVersion: version,
```

Comment:
we may need to add some more fields from what was recently added such as `GitHash` and `Env`. Don't think there's any others that are applicable from the list.